File: 1/e/e.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Cell {
    int x; 
    int y;
}Cell;

void main() {
    int m;
    int n; 
    int area_torre = 0; 

    int h; 
    int acc = 0;

    scanf("%d %d", &n, &m);
    Cell tower[n*m];
    char field[n][m], aux;

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++)  {
            scanf("%c", &aux);
            if  (aux == '\n') 
                scanf("%c", &aux);
            field[i][j] = aux;
            if  (aux == 't') 
                tower[area_torre++] = (Cell){i, j};
        }
    }


    for (int i = 0; i < area_torre; i++) {
        scanf("%d", &h);
        for (int j = tower[i].x - h; j <= tower[i].x + h; j++)
            for (int k = tower[i].y - h; k <= tower[i].y + h; k++)
                if (j >= 0 && j < n && k >= 0 && k < m && field[j][k] == '#') {
                    acc++;
                    field[j][k] = '.';
                }
    }
    printf("%d\n", acc);
    for (int i = 0; i < n; i++) {
        for(int j = 0; j < m; j++)
            printf("%c", field[i][j]);        
        printf("\n");
    }
}

File: 1/e/e.go
package main

import (
	"fmt" )

type Item struct {
    x  int
    y  int
}

func main() {

   var n, m int
   var h int
   var tower_height = 0
   var acc = 0

   fmt.Scanf("%d %d", &n, &m)

   var tower = []Item{}
   //var field [][]rune
   field := make([][]rune, n, m)
   var aux rune
   for i := 0; i < n; i++{
	for j := 0; j < m; j++ {
		fmt.Scanf("%c", &aux)
		if aux == '\n'{
			fmt.Scanf("%c", &aux);
		}
		field[i] = append(field[i], aux)
		if aux == 't'{
		 	tower = append(tower, Item{x: i, y: j})
			tower_height++
		 }
		}
	}

	fmt.Println(tower)
	for i := 0; i < n; i++{
		for j := 0; j < m; j++ {
			fmt.Printf("%c", field[i][j])
		}
		fmt.Println()
	}

	for i := 0; i < tower_height; i++{
		fmt.Scanf("%d", &h)
		min_x := tower[i].x - h 
		max_x := tower[i].x + h
		for j := min_x; j <= max_x; j++{
			min_y := tower[i].y - h
			max_y := tower[i].y + h
			for k := min_y; k <= max_y; k++ {
				if j >= 0 && j < n && k >= 0 && k < m && field[j][k] == '#'{
					acc++
					field[j][k] = '.'
				} 
			}

		}
	}


	fmt.Println(acc)

	for i := 0; i < n; i++{
		for j := 0; j < m; j++ {
			fmt.Printf("%c", field[i][j])
		}
		fmt.Println()
	}



}